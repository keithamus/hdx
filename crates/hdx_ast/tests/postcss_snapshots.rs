mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
