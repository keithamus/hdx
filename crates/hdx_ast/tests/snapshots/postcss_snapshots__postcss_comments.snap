---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Html": {
                  "Div": {
                    "kind": "Ident",
                    "offset": 36,
                    "len": 3
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 40,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 63,
            "len": 5
          },
          "colon": {
            "kind": "Colon",
            "offset": 68,
            "len": 1
          },
          "value": {
            "Named": {
              "black": {
                "kind": "Ident",
                "offset": 70,
                "len": 5
              }
            }
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 75,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 99,
            "len": 13
          },
          "colon": {
            "kind": "Colon",
            "offset": 112,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 113,
                "len": 1
              },
              {
                "kind": "Dimension",
                "offset": 114,
                "len": 3,
                "unit": "px"
              },
              {
                "kind": "Whitespace",
                "offset": 117,
                "len": 1
              },
              {
                "kind": "Delim",
                "offset": 118,
                "len": 1
              },
              {
                "kind": "Whitespace",
                "offset": 119,
                "len": 1
              },
              {
                "kind": "Dimension",
                "offset": 120,
                "len": 3,
                "unit": "px"
              }
            ]
          },
          "important": null,
          "semicolon": null
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 138,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Html": {
                  "A": {
                    "kind": "Ident",
                    "offset": 149,
                    "len": 1
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 151,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 157,
            "len": 5
          },
          "colon": {
            "kind": "Colon",
            "offset": 162,
            "len": 1
          },
          "value": {
            "Named": {
              "black": {
                "kind": "Ident",
                "offset": 164,
                "len": 5
              }
            }
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 169,
            "len": 1
          }
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 183,
        "len": 1
      }
    },
    {
      "type": "QualifiedRule",
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 184,
            "len": 2
          },
          {
            "kind": "Delim",
            "offset": 186,
            "len": 1
          },
          {
            "kind": "Ident",
            "offset": 187,
            "len": 2
          },
          {
            "kind": "Ident",
            "offset": 193,
            "len": 4
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 197,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 198,
          "len": 1
        },
        "declarations": [
          {
            "type": "Declaration",
            "name": {
              "kind": "Ident",
              "offset": 202,
              "len": 5
            },
            "colon": {
              "kind": "Colon",
              "offset": 207,
              "len": 1
            },
            "value": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 208,
                  "len": 1
                },
                {
                  "kind": "Ident",
                  "offset": 209,
                  "len": 3
                },
                {
                  "type": "SimpleBlock",
                  "start": 212,
                  "open": {
                    "kind": "LeftParen",
                    "len": 1
                  },
                  "values": {
                    "values": [
                      {
                        "kind": "Ident",
                        "offset": 217,
                        "len": 8
                      }
                    ]
                  },
                  "close": {
                    "kind": "RightParen",
                    "len": 1
                  }
                }
              ]
            },
            "important": null,
            "semicolon": {
              "kind": "Semicolon",
              "offset": 226,
              "len": 1
            }
          },
          {
            "type": "Declaration",
            "name": {
              "kind": "Ident",
              "offset": 230,
              "len": 11
            },
            "colon": {
              "kind": "Colon",
              "offset": 241,
              "len": 1
            },
            "value": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 242,
                  "len": 1
                },
                {
                  "kind": "Ident",
                  "offset": 243,
                  "len": 10
                },
                {
                  "kind": "Comma",
                  "offset": 253,
                  "len": 1
                },
                {
                  "kind": "Ident",
                  "offset": 258,
                  "len": 5
                }
              ]
            },
            "important": null,
            "semicolon": {
              "kind": "Semicolon",
              "offset": 263,
              "len": 1
            }
          }
        ],
        "rules": [],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 265,
          "len": 1
        }
      }
    }
  ]
}
